server/src/internalClusterTest/java/org/elasticsearch/recovery/FullRollingRestartIT.java

```diff
@@ -201,7 +200,14 @@ public void testNoRebalanceOnRollingRestart() throws Exception {
         RecoveryResponse recoveryResponse = client().admin().indices().prepareRecoveries("test").get();
         for (RecoveryState recoveryState : recoveryResponse.shardRecoveryStates().get("test")) {
             assertTrue(
-                "relocated from: " + recoveryState.getSourceNode() + " to: " + recoveryState.getTargetNode() + "\n" + state,
+                "relocated "
+                    + recoveryState.getShardId()
+                    + " from: "
+                    + recoveryState.getSourceNode()
+                    + " to: "
+                    + recoveryState.getTargetNode()
+                    + "\n"
+                    + state,
                 recoveryState.getRecoverySource().getType() != RecoverySource.Type.PEER || recoveryState.getPrimary() == false
```

Comment:
I believe `recoveryState.getPrimary() == false` could be removed since all shards in this test are primaries.
It would be also great to report all shard movements, not just the first one in case of failure